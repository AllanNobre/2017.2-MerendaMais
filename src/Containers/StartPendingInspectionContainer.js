import { connect } from 'react-redux';
import StartPendingInspection from '../screens/StartPendingInspection';
import { asyncGetSchedule } from '../actions/schedulingActions';

const mapStateToProps = state => ({
  counselor: state.counselor,
  listOfPendingScheduleInAGroup: state.list.listOfPendingScheduleInAGroup,
  listOfExpiredScheduleInAGroup: state.list.listOfExpiredScheduleInAGroup,
  listOfAlreadyInpectionedSchedueInAGroup:
   state.list.listOfAlreadyInpectionedSchedueInAGroup,
});

const mapDispatchToProps = dispatch => ({
  asyncGetSchedule: counselor => dispatch(asyncGetSchedule(counselor)),
});

const StartPendingInspectionContainer = connect(mapStateToProps,
  mapDispatchToProps)(StartPendingInspection);

export default StartPendingInspectionContainer;
