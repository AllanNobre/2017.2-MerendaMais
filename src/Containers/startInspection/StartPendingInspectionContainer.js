import { connect } from 'react-redux';
import StartPendingInspection from '../../screens/startInspection/StartPendingInspection';
import { asyncGetSchedule } from '../../actions/schedulingVisitActions';
import { asyncGetCounselorFromGroup } from '../../actions/listActions';

const mapStateToProps = state => ({
  counselor: state.counselor,
  listOfPendingScheduleInAGroup: state.list.listOfPendingScheduleInAGroup,
  listOfCounselorsInAGroup: state.list.listOfCounselorsInAGroup,
});

const mapDispatchToProps = dispatch => ({
  asyncGetSchedule: counselor => dispatch(asyncGetSchedule(counselor)),
  asyncGetCounselorFromGroup: (CAE, CPF) => dispatch(asyncGetCounselorFromGroup(CAE, CPF)),
});

const StartPendingInspectionContainer = connect(mapStateToProps,
  mapDispatchToProps)(StartPendingInspection);

export default StartPendingInspectionContainer;
