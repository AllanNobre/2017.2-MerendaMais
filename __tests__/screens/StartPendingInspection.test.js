import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// import configureStore from 'redux-mock-store';
import StartPendingInspectionContainer from '../../src/Containers/startInspection/StartPendingInspectionContainer';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  application: {
    isLoading: false,
  },

  list: {
    listOfPendingScheduleInAGroup: {
      codSchool: '',
      date: '',
      time: '',
    },
    listOfCounselorsInAGroup: {

    },
  },

  counselor: {

  },

  asyncGetSchedule: jest.fn(),
  asyncGetCounselorFromGroup: jest.fn(),
};

const store = mockStore(initialState);

describe('Testing StartPendingInspection Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <StartPendingInspectionContainer />,
      { context: { store } },
    );
    expect(wrapper).toMatchSnapshot();
  });
});
