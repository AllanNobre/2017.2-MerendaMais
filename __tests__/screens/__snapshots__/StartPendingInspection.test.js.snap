// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StartPendingInspection Screen renders as expected 1`] = `
<StartPendingInspection
  asyncGetCounselorFromGroup={[Function]}
  asyncGetSchedule={[Function]}
  counselor={Object {}}
  isLoading={false}
  listOfCounselorsInAGroup={Object {}}
  listOfPendingScheduleInAGroup={
    Object {
      "codSchool": "",
      "date": "",
      "time": "",
    }
  }
/>
`;
